### ISSUE TITLE
Option to enable touch ID authentication for sudo doesn't work in tmux

### ISSUE THREAD

**Issue by 347Online** [LnL7/nix-darwin#985]
I really love the out-of-the-box option to enable touch ID for sudo and was surprised to find it doesn't work in tmux

I eventually stumbled across [this](https://github.com/fabianishere/pam_reattach), which is working for me, but I know it's going to get blown away when macOS does an update

I spent awhile noodling around with getting this working in my personal config to no avail, but it'd be even better to just have this as an option in nix-darwin — programs.pam-reattach.enable perhaps

**Comment by emilazy** [comment]
Yeah, finally getting this in with the rework of the sudo PAM stuff is on my to‐do list. See also: https://github.com/LnL7/nix-darwin/pull/787, https://github.com/LnL7/nix-darwin/pull/662, https://github.com/LnL7/nix-darwin/pull/606, https://github.com/LnL7/nix-darwin/pull/591. (This is probably the area of nix-darwin that has evolved the slowest for how many PRs it’s had…)

My current plan is as outlined in https://github.com/LnL7/nix-darwin/pull/787#issuecomment-2165246619.

**Comment by 347Online** [comment]
Ahh, thank you that's fairly illuminating! My apologies if I've created extra noise here, for whatever reason I was not able to turn up any of these prior conversations when I searched

**Comment by emilazy** [comment]
No worries, I just wish I’d made progress on this sooner!

**Comment by gshpychka** [comment]
FWIW this works for me: https://github.com/gshpychka/dotfiles/blob/140600b6b4cec63b36f9a94d04fcd6e2f65251ee/machines/eve/touch-id.nix

You don't need `pam_watchid` - I believe that was due to a quirk of pam_tid not working with a closed lid

**Comment by michaelvanstraten** [comment]
> FWIW this works for me: https://github.com/gshpychka/dotfiles/blob/140600b6b4cec63b36f9a94d04fcd6e2f65251ee/machines/eve/touch-id.nix
> 
> You don't need `pam_watchid` - I believe that was due to a quirk of pam_tid not working with a closed lid

This bricked my system, I had to reboot into recovery mode to undo this change.

**Comment by Enzime** [comment]
Closed by #1344
